src/main/java/uk/ac/ebi/atlas/search/metadata/MetadataSearchDao.java

```diff
@@ -0,0 +1,92 @@
+package uk.ac.ebi.atlas.search.metadata;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.download.FileDownloadController;
+import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
+import uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy;
+import uk.ac.ebi.atlas.solr.cloud.search.SolrQueryBuilder;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CELL_ID;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CHARACTERISTIC_NAME;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CHARACTERISTIC_VALUE;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.EXPERIMENT_ACCESSION;
+
+@Component
+public class MetadataSearchDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileDownloadController.class);
```

Comment:
Please change this to `MetadataSearchDao.class` in `getLogger`

Reply:
As we are dealing with backend(Kind of data repository) and we mentioned DAO as part of the class name. So I would feel that @Repository  annotation suits more here than just independent @Component annotation.

Reply:
In the search package, I could see 'GeneSearchDao' class. It is clearly informing the developer that this class deals with only genes. If we have any intention of that this class deals with only Cell metadata. We can rename the class to 'CellMetadataDao'

Reply:
Just to clear things out a bit, we’ve never annotated DAO classes with `@Repository` because we’re not (or maybe it’s just me?) big fans of DDD. More seriously, since we’re not following any DDD principles because we’re old-school I can’t see the benefit of bothering with `@Component` subtypes (e.g. `@Service`), and they receive no special treatment. @upendrakumbham Is there any benefit of using `@Repository` (we’re no big Spring specialists)?

Reply:
Since ages, I have got a habit of using @Component annotation for -  any Spring-managed component which doesn't belong to presentation, service, and data access layers. Somewhere I read about this in the spring documentation when I was learning spring. They would recommend using the more suitable stereotype annotations to improve readability as well as some special benefits in future releases for those annotations.
Yes, @Repository annotation has some special benefits to use.
[Stackoverflow Details](https://stackoverflow.com/questions/6827752/whats-the-difference-between-component-repository-service-annotations-in)
 @Component and @Service - currently both are same -  component scan and register a bean with the application context.
But we don't know, spring team may add some weightage to the @Service as well in the future releases :).

Reply:
I see! Thanks for the links. We can add `@Repository` to all of our DAO classes (as we go about adding and updating them, no rush to do it), with the understanding that in our project DAO and DDD repositories are pretty much the same.